packages/opentelemetry-core/test/context/B3Propagator.test.ts

```diff
@@ -98,127 +157,221 @@ describe('B3Propagator', () => {
+
+    it('should extract context of a span from carrier when debug flag is invalidly set as 0', () => {
+      carrier[X_B3_TRACE_ID] = '0af7651916cd43dd8448eb211c80319c';
+      carrier[X_B3_SPAN_ID] = 'b7ad6b7169203331';
+      carrier[X_B3_FLAGS] = '0';
+      carrier[X_B3_SAMPLED] = '1';
+      const context = b3Propagator.extract(
+        Context.ROOT_CONTEXT,
+        carrier,
+        defaultGetter
+      );
+      const extractedSpanContext = getExtractedSpanContext(context);
+
+      assert.deepStrictEqual(extractedSpanContext, {
+        spanId: 'b7ad6b7169203331',
+        traceId: '0af7651916cd43dd8448eb211c80319c',
+        isRemote: true,
+        traceFlags: TraceFlags.SAMPLED,
+      });
+      assert.equal(context.getValue(DEBUG_FLAG_KEY), undefined);
+      assert.equal(context.getValue(PARENT_SPAN_ID_KEY), undefined);
+    });
+
+    it('should extract context of a span from carrier with a parent span id', () => {
+      carrier[X_B3_TRACE_ID] = '0af7651916cd43dd8448eb211c80319c';
+      carrier[X_B3_SPAN_ID] = 'b7ad6b7169203331';
+      carrier[X_B3_PARENT_SPAN_ID] = 'f4592dc481026a8c';
+      carrier[X_B3_FLAGS] = '0';
+      carrier[X_B3_SAMPLED] = '1';
+      const context = b3Propagator.extract(
+        Context.ROOT_CONTEXT,
+        carrier,
+        defaultGetter
+      );
+      const extractedSpanContext = getExtractedSpanContext(context);
+
+      assert.deepStrictEqual(extractedSpanContext, {
+        spanId: 'b7ad6b7169203331',
+        traceId: '0af7651916cd43dd8448eb211c80319c',
+        isRemote: true,
+        traceFlags: TraceFlags.SAMPLED,
+      });
+      assert.equal(context.getValue(DEBUG_FLAG_KEY), undefined);
+      assert.equal(context.getValue(PARENT_SPAN_ID_KEY), 'f4592dc481026a8c');
     });
 
     it('should return undefined when traceId is undefined', () => {
       carrier[X_B3_TRACE_ID] = undefined;
       carrier[X_B3_SPAN_ID] = undefined;
-      assert.deepStrictEqual(
-        getExtractedSpanContext(
-          b3Propagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
-        ),
-        undefined
+      const context = getExtractedSpanContext(
+        b3Propagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
       );
+      assert.deepStrictEqual(context, undefined);
     });
 
-    it('should return undefined when options and spanId are undefined', () => {
+    it('should return undefined when spanId is undefined', () => {
       carrier[X_B3_TRACE_ID] = '0af7651916cd43dd8448eb211c80319c';
       carrier[X_B3_SPAN_ID] = undefined;
-      assert.deepStrictEqual(
-        getExtractedSpanContext(
-          b3Propagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
-        ),
-        undefined
+      const context = getExtractedSpanContext(
+        b3Propagator.extract(Context.ROOT_CONTEXT, carrier, defaultGetter)
       );
+      assert.deepStrictEqual(context, undefined);
+    });
+
+    it('should return undefined when parentSpanId is invalid', () => {
```

Comment:
I think if parentSpanId is invalid, I expect it should at-least extract context of a span from carrier w/o a parent_span_id. WDYT?

Reply:
Umm, I think I saw an implementation of this in go and js by Zipkin, and I think they just threw away everything. Let me double-check.

Reply:
Yeah, they just throw it away: https://github.com/openzipkin/zipkin-go/blob/master/propagation/b3/spancontext.go#L83

Reply:
Hmm, they also seem to throw everything out if the sampling number isn't right, but debug doesn't matter.

Reply:
They also allow for `undefined` sampling, which we don't define in span context. Should I handle all that in a new issue?

Reply:
> Umm, I think I saw an implementation of this in go and js by Zipkin, and I think they just threw away everything. Let me double-check.

One possible option is to log an error message and move on. @dyladan @obecny WDYT?

Reply:
Got a response: https://github.com/openzipkin/b3-propagation/issues/41. If I'm reading this right, sounds like we shouldn't try to handle this ourselves and just move on.

Reply:
So, I changed the `spanContext.traceFlags` to be optional, and I made the extract strict. I we want to change how strict it is, we just erase from the condition and remove the tests.

Reply:
According to adrian in that thread, brave drops the trace and restarts on an invalid/undefined behavior. This is also the behavior for malformed trace ids/flags in the W3C spec and implementation. I would say for consistency it may be better to do that.

Reply:
Cool, then that's what I'm doing, at least I think because looking at the httpTraceContext propagator, it just returns the context.